apache/superset
# feat(dashboard_rbac): dashboard lists

### SUMMARY
Role based dashboard level access with focus on dashboard list APIs

credits also goes to @Ofeknielsen who took [part](https://github.com/Ofeknielsen/incubator-superset/tree/feat/dashboard_permission_enforcement/) in this months ago 


resolves #10408 
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/47772523/105577918-39ff5380-5d85-11eb-987b-2c3a3450480a.mov

### Design slide;
https://user-images.githubusercontent.com/4025227/105494222-6302e400-5cb2-11eb-8f0d-6a609d73cc6c.png

### TEST PLAN

ff:
DASHBOARD_RBAC: True
ENABLE_REACT_CRUD_VIEWS: False

happy path:
1. create a new user X with Gamma role
2. edit a dashboard and add to roles the Gamma role
3. login as X and see the assigned dashboard

backward compatibility:
check backward compatibility by DASHBOARD_RBAC: False 
access should be managed by the data level

#### code test coverage plan
dashboards list page for the following scenarios both for dataset and RBAC levels
To make sure we keep backward compatibility and new feature alignment on capabilities

1. an anonymous user
2. a non-admin (test user: alpha/general)
3. admin
4. owner
5. Published/ non published 

Test need to be written in a decoupled approch from existing users/roles/tables/dashboards so other tests suites won't break 

### ADDITIONAL INFORMATION
